packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -576,33 +653,14 @@ class RenderListWheelViewport
     }
   }
 
-  /// Visits all the children until one is partially visible in the viewport.
-  RenderBox _getFirstVisibleChild() {
-    assert(childCount > 0);
-    final double firstVisibleLayoutOffset = _firstVisibleLayoutOffset;
-
-    RenderBox child = firstChild;
-    ListWheelParentData childParentData = child.parentData;
-
-    while (childParentData != null
-        && childParentData.offset.dy <= firstVisibleLayoutOffset) {
-      child = childAfter(child);
-      childParentData = child?.parentData;
-    }
-
-    return child;
-  }
-
   /// Paints all children visible in the current viewport.
   void _paintVisibleChildren(PaintingContext context, Offset offset) {
     assert(childCount > 0);
-    final double lastVisibleLayoutOffset = _lastVisibleLayoutOffset;
 
-    RenderBox childToPaint = _getFirstVisibleChild();
+    RenderBox childToPaint = firstChild;
     ListWheelParentData childParentData = childToPaint?.parentData;
 
-    while (childParentData != null
-        && childParentData.offset.dy < lastVisibleLayoutOffset) {
```

Comment:
Does `clipToSize = false`, ` renderChildrenOutsideViewport = true` still work?

Reply:
No, I don't think so.
Now I only keep the the children that are actually in visible range in the child list, and remove all children that are out of range. This implies that even if the picker is constructed from a concrete list of children, the actual child list of the render viewport can be smaller.
I don't see any usage of rendering children outside viewport yet, is it important? Can you explain more?

Reply:
It's not, we should just be very careful when we break public APIs. Look on GitHub for any occurrences of usages. If it's going to be hard to support, we'd need to make a public announcement that we're breaking this API and remove these parameters.

Reply:
These parameters are now problematic because when we rely on the childManager, we can't know for sure how many children there are and then we have to decide how many children we want to paint outside the viewport, etc.
Since this feature is not being used anywhere (I have checked) and I don't feel like it is important, can we remove it?

Reply:
Ya, sure, remove it. Just make sure to check (in open source code) if it's used on the internet and send an email to the flutter-dev googlegroup (see other announcement emails for example)

Reply:
I have fixed the code a little bit so that `renderChildrenOutsideViewport` can work, by spawning more children. Please check!
If the code seems good to you, then I would rather keep the parameter, so that nothing will get broken : )